src/PreCharge/GFDB.cpp

```diff
@@ -0,0 +1,121 @@
+#include <PreCharge/GFDB.hpp>
+
+namespace IO = EVT::core::IO;
+namespace time = EVT::core::time;
+
+namespace GFDB {
+
+GFDB::GFDB(IO::CAN& can) : can(can){};
+
+IO::CAN::CANStatus GFDB::init() {
+    return can.connect();
+}
+
+IO::CAN::CANStatus GFDB::requestVnHighRes(int32_t* highRes) {
+    uint8_t rxBuffer[8] = {};
+    IO::CAN::CANStatus result = requestData(0x60, rxBuffer, 8);
+    if (result == IO::CAN::CANStatus::ERROR)
+        return result;
+
+    *highRes = (rxBuffer[0] << 24) | (rxBuffer[1] << 16) | (rxBuffer[2] << 8) | rxBuffer[3];
+    return IO::CAN::CANStatus::OK;
+}
+
+IO::CAN::CANStatus GFDB::requestVpHighRes(int32_t* highRes) {
+    uint8_t rxBuffer[8] = {};
+    IO::CAN::CANStatus result = requestData(0x61, rxBuffer, 8);
+    if (result == IO::CAN::CANStatus::ERROR)
+        return result;
+
+    *highRes = (rxBuffer[0] << 24) | (rxBuffer[1] << 16) | (rxBuffer[2] << 8) | rxBuffer[3];
```

Comment:
What is this doing? It is a lot of logical operations in one line and I think it could be separated out.

Reply:
This is just a standard way of converting an array of 8 bit integers to 32 bit